
This patch adds Quake packages to buildroot:
* Quake engine itself (SDL port)
* a shareware data file (PAK/WAD) to test the engine

It is inspired of OpenEmbedded work.

Still searching a way to apply sdlquake-1.0.9-no-x86-asm.patch on all arch
EXCEPT x86.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: buildroot/package/sdlquake/sdlquake.mk
===================================================================
--- /dev/null
+++ buildroot/package/sdlquake/sdlquake.mk
@@ -0,0 +1,22 @@
+#############################################################
+#
+# Quake 1 (SDL Port)
+#
+#############################################################
+
+SDLQUAKE_VERSION = 1.0.9
+SDLQUAKE_SITE = http://www.libsdl.org/projects/quake/src/
+SDLQUAKE_AUTORECONF = YES
+
+SDLQUAKE_CONF_OPTS = --with-sdl-prefix=$(STAGING_DIR)/usr
+
+SDLQUAKE_DEPENDENCIES = sdl
+
+define SDLQUAKE_AUTOCONF_FIXES
+	(cd $(@D); touch INSTALL NEWS README AUTHORS ChangeLog; \
+		echo "AC_PROG_CC" >> configure.in;)
+endef
+
+SDLQUAKE_POST_PATCH_HOOKS += SDLQUAKE_AUTOCONF_FIXES
+
+$(eval $(autotools-package))
Index: buildroot/package/sdlquake/sdlquake-1.0.9-no-x86-asm.patch.arm
===================================================================
--- /dev/null
+++ buildroot/package/sdlquake/sdlquake-1.0.9-no-x86-asm.patch.arm
@@ -0,0 +1,35 @@
+--- /tmp/Makefile.am	2008-08-11 20:31:09.000000000 +0200
++++ sdlquake-1.0.9/Makefile.am	2008-08-11 20:55:00.263198000 +0200
+@@ -33,7 +33,6 @@
+ 	crc.h			\
+ 	cvar.c			\
+ 	cvar.h			\
+-	d_copy.S		\
+ 	d_edge.c		\
+ 	d_fill.c		\
+ 	d_iface.h		\
+@@ -48,7 +47,6 @@
+ 	d_sprite.c		\
+ 	d_surf.c		\
+ 	d_zpoint.c		\
+-	dosasm.S		\
+ 	dosisms.h		\
+ 	draw.c			\
+ 	draw.h			\
+@@ -103,7 +101,6 @@
+ 	r_sprite.c		\
+ 	r_surf.c		\
+ 	r_vars.c		\
+-	r_varsa.S		\
+ 	render.h		\
+ 	resource.h		\
+ 	sbar.c			\
+@@ -136,7 +133,7 @@
+ 	world.h			\
+ 	zone.c			\
+ 	zone.h			\
+-	$(X86_SRCS) $(NONX86_SRCS)
++	$(NONX86_SRCS)
+
+ X86_SRCS =			\
+ 	snd_mixa.S		\
Index: buildroot/package/sdlquake/Config.in
===================================================================
--- /dev/null
+++ buildroot/package/sdlquake/Config.in
@@ -0,0 +1,10 @@
+config BR2_PACKAGE_SDLQUAKE
+	bool "sdlquake"
+	select BR2_PACKAGE_SDL
+	select BR2_PACKAGE_SDL_MIXER
+	select BR2_PACKAGE_SDL_NET
+	help
+	  sdlquake is a port of ID Software's popular Quake 3-D shooter game.
+	  (Not supported by ID Software)
+
+	  http://www.libsdl.org/projects/quake/
Index: buildroot/package/quake-pak/quake-pak.mk
===================================================================
--- /dev/null
+++ buildroot/package/quake-pak/quake-pak.mk
@@ -0,0 +1,34 @@
+#############################################################
+#
+# Quake shareware data files (PAK/WAD)
+#
+#############################################################
+
+QUAKE_PAK_VERSION = 1.0.6
+QUAKE_PAK_SOURCE = quakesw-$(QUAKE_PAK_VERSION).tar.gz
+QUAKE_PAK_SITE = http://www.libsdl.org/projects/quake/data/
+QUAKE_PAK_CAT = $(ZCAT)
+QUAKE_PAK_TARGET_DIR = $(TARGET_DIR)/usr/share/games/quake/
+
+$(DL_DIR)/$(QUAKE_PAK_SOURCE):
+	 $(WGET) -P $(DL_DIR) $(QUAKE_PAK_SITE)/$(QUAKE_PAK_SOURCE)
+
+quake-pak-source: $(DL_DIR)/$(QUAKE_PAK_SOURCE)
+
+$(QUAKE_PAK_TARGET_DIR)/id1/pak0.pak: $(DL_DIR)/$(QUAKE_PAK_SOURCE)
+	mkdir -p $(QUAKE_PAK_TARGET_DIR)
+	$(QUAKE_PAK_CAT) $(DL_DIR)/$(QUAKE_PAK_SOURCE) | tar -C $(QUAKE_PAK_TARGET_DIR) $(TAR_OPTIONS) -
+
+quake-pak: $(QUAKE_PAK_TARGET_DIR)/id1/pak0.pak
+
+quake-pak-dirclean:
+	rm -rf $(QUAKE_PAK_TARGET_DIR)/id1/
+
+#############################################################
+#
+# Toplevel Makefile options
+#
+#############################################################
+ifeq ($(strip $(BR2_PACKAGE_QUAKE_PAK)),y)
+TARGETS+=quake-pak
+endif
Index: buildroot/package/quake-pak/Config.in
===================================================================
--- /dev/null
+++ buildroot/package/quake-pak/Config.in
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_QUAKE_PAK
+	bool "shareware Quake PAK/WAD file"
+	depends on BR2_PACKAGE_SDLQUAKE
+	help
+	  This will install the shareware pak/wad data file for the Quake game.
+
+	  The file will be placed in the /usr/share/games/quake/ directory.
Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in
+++ buildroot/package/Config.in
@@ -184,7 +184,9 @@ menu "Games"
	source "package/opentyrian/Config.in"
	source "package/opentyrian-data/Config.in"
	source "package/prboom/Config.in"
+	source "package/quake-pak/Config.in"
	source "package/rubix/Config.in"
+	source "package/sdlquake/Config.in"
 endmenu

 menu "Graphic libraries and applications (graphic/text)"
